tasks/src/tasks/append.rs

```diff
@@ -0,0 +1,175 @@
+use imap_types::{
+    command::CommandBody,
+    datetime::DateTime,
+    extensions::binary::LiteralOrLiteral8,
+    flag::Flag,
+    mailbox::Mailbox,
+    response::{Data, StatusBody, StatusKind},
+};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug)]
+pub struct AppendTask {
+    mailbox: Mailbox<'static>,
+    flags: Vec<Flag<'static>>,
+    date: Option<DateTime>,
+    message: LiteralOrLiteral8<'static>,
+    output: Option<u32>,
+}
+
+impl Task for AppendTask {
+    type Output = Result<Option<u32>, TaskError>;
+
+    fn command_body(&self) -> CommandBody<'static> {
+        CommandBody::Append {
+            mailbox: self.mailbox.clone(),
+            flags: self.flags.clone(),
+            date: self.date.clone(),
+            message: self.message.clone(),
+        }
+    }
+
+    fn process_data(&mut self, data: Data<'static>) -> Option<Data<'static>> {
+        // In case the mailbox is already selected, we should receive
+        // an `EXISTS` response.
+        if let Data::Exists(seq) = data {
+            self.output = Some(seq);
```

Comment:
Hm... duplicates again. Maybe we should think about a more generic solution. For now, log should suffice?